### PR TITLE
Doc/update cli help msg

### PR DESCRIPTION
**Pull request by zhenyu-ms**
## Bug / Requirement Description
Clearly and concisely describe the problem.

## Solution description
Describe your code changes in detail for reviewers.

## Checklist:
- [ ] Test
- [ ] Example (both test_plan.py and .rst)
- [x] Documentation (API)
- [ ] News fragment present for release notes
- [x] MS info leakage check
- [ ] For new driver: driver index page
- [ ] For new assertion: ui/pdf/std renderers, documentation
- [ ] For new cmdline arg: documentation

### CODE REVIEW

File: doc/en/introduction.rst [hunk 1]
```diff
@@ -350,7 +350,14 @@ Command line
                                 ......
                             }
       --skip-remaining {cases-on-failed,cases-on-error,suites-on-failed,suites-on-error,tests-on-failed,tests-on-error}
-                            Skip the remaining Testcases/Testsuites/Multitests being executed when a Testcase has failed or raised exception.
+                            Make Testplan skip certain testcases from execution upon exception ("error") or assertion failed ("failed"). Here "failed" includes "error" situation.
+
+                            Use "cases-on-failed"/"cases-on-error" to skip remaining testcases in the same testsuite when condition meets.
```
Comment (Pyifan) [inline doc/en/introduction.rst:355]: and resume with the next testsuite.

File: doc/en/introduction.rst [hunk 2]
```diff
@@ -350,7 +350,14 @@ Command line
...
+
+                            Use "suites-on-failed"/"suites-on-error" to skip remaining testsuites as well in the same Multitest when condition meets.
```
Comment (Pyifan) [inline doc/en/introduction.rst:357]: but resume with other tests (GTest, Multitest).

File: doc/en/introduction.rst [hunk 3]
```diff
@@ -350,7 +350,14 @@ Command line
                                 ......
                             }
       --skip-remaining {cases-on-failed,cases-on-error,suites-on-failed,suites-on-error,tests-on-failed,tests-on-error}
-                            Skip the remaining Testcases/Testsuites/Multitests being executed when a Testcase has failed or raised exception.
+                            Make Testplan skip certain testcases from execution upon exception ("error") or assertion failed ("failed"). Here "failed" includes "error" situation.
```
Comment (Pyifan) [inline doc/en/introduction.rst:353]: This option makes testplan to break from the current execution flow on error/failed and skip remaining iterations at certain level.

When on-error is specified, skip-remaining is triggered when exception("error") happens during testing. When on-failed is specified, skip-remaining is triggered upon both exception("error") and test failure("failed").

To --skip-remaining cases, means skip remaining testcases in the same testsuite, and resume with next testsuite.

Comment (Pyifan) [reply doc/en/introduction.rst:353]: To --skip-remaining suites, means skip remaining testcases in the current testsuite, and remaining testsuites in the current Multitest, and resume with next Multitest.

To --skip-remaining tests, means stop everything - remaining testcases/testsuites in all the Multitests that are currently executing, and remaining tests that has not yet start executing.

File: doc/en/introduction.rst [hunk 4]
```diff
@@ -350,7 +350,14 @@ Command line
                                 ......
                             }
       --skip-remaining {cases-on-failed,cases-on-error,suites-on-failed,suites-on-error,tests-on-failed,tests-on-error}
-                            Skip the remaining Testcases/Testsuites/Multitests being executed when a Testcase has failed or raised exception.
+                            Make Testplan break from the current execution flow and skip remaining iterations at certain level. "on-error" make this skip triggered upon exception raised, and "on-failed" make this skip triggered upon both exception raised and test failure.
+
+                            Use "cases-on-failed"/"cases-on-error" to skip remaining testcases in the same testsuite when condition meets, execution will resume from next from next testsuite.
```
Comment (M6AI) [inline doc/en/introduction.rst:355]:
```suggestion
                            Use "cases-on-failed"/"cases-on-error" to skip remaining testcases in the same testsuite when condition is met, the execution will resume from the next testsuite.
```

File: doc/en/introduction.rst [hunk 5]
```diff
@@ -350,7 +350,14 @@ Command line
...
+
+                            Use "suites-on-failed"/"suites-on-error" to skip remaining testsuites as well in the same Multitest when condition meets, execution will resume from next Multitest/GTest etc.
```
Comment (M6AI) [inline doc/en/introduction.rst:357]:
```suggestion
                            Use "suites-on-failed"/"suites-on-error" to skip remaining testsuites as well in the same Multitest when condition is met, execution will resume from the next Multitest/GTest etc.
```

File: doc/en/introduction.rst [hunk 6]
```diff
@@ -350,7 +350,14 @@ Command line
...
+
+                            Use "tests-on-failed"/"tests-on-error" to skip remaining Multitests/GTests etc. as well (i.e. everything remaining) in the current Testplan when condition meets.
```
Comment (M6AI) [inline doc/en/introduction.rst:359]:
```suggestion
                            Use "tests-on-failed"/"tests-on-error" to skip remaining Multitests/GTests etc. as well (i.e. everything remaining) in the current Testplan when condition is met.
```

File: testplan/parser.py [hunk 7]
```diff
@@ -218,8 +218,19 @@ def generate_parser(self) -> HelpParser:
...
+            "exception raised and test failure.\n"
+            'Use "cases-on-failed"/"cases-on-error" to skip remaining testcases in the '
+            "same testsuite when condition meets, execution will resume from next from "
+            "next testsuite.\n"
+            'Use "suites-on-failed"/"suites-on-error" to skip remaining testsuites as '
+            "well in the same Multitest when condition meets, execution will resume "
+            "from next Multitest/GTest etc.\n"
+            'Use "tests-on-failed"/"tests-on-error" to skip remaining Multitests/GTests '
+            "etc. as well (i.e. everything remaining) in the current Testplan when "
+            "condition meets.\n",
```
Comment (M6AI) [inline testplan/parser.py:233]: Just update accordingly as the comments above.

**Review by M6AI — DISMISSED** [review]

**Review by kelliott55 — APPROVED** [review]

**Review by kelliott55 — APPROVED** [review]

**Pull request merged**